src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -39,6 +39,38 @@ public CertificateCallbackMapper(Func<HttpRequestMessage, X509Certificate2, X509
         }
 
         public static async ValueTask<(Socket, Stream)> ConnectAsync(string host, int port, CancellationToken cancellationToken)
+        {
+            CancellationTokenSource cancelIPv6 = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            Task<(Socket, Stream)> tryConnectAsyncIPv6 = ConnectAsyncInternal(host, port, AddressFamily.InterNetworkV6, cancelIPv6.Token);
+
+            if (await Task.WhenAny(tryConnectAsyncIPv6, Task.Delay(200)).ConfigureAwait(false) == tryConnectAsyncIPv6 && tryConnectAsyncIPv6.IsCompletedSuccessfully)
+            {
+                return await tryConnectAsyncIPv6.ConfigureAwait(false);
+            }
+
+            CancellationTokenSource cancelIPv4 = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            Task<(Socket, Stream)> tryConnectAsyncIPv4 = ConnectAsyncInternal(host, port, AddressFamily.InterNetwork, cancelIPv4.Token);
+            if (await Task.WhenAny(tryConnectAsyncIPv6, tryConnectAsyncIPv4).ConfigureAwait(false) == tryConnectAsyncIPv6)
```

Comment:
Task.WhenAny takes a params array, which means both of these Task.WhenAny calls is allocating a Task[2].  You could instead allocate one once and pass that same array into both calls, just populating Task[1] differently for each.